src/connector/src/dummy_connector.rs

```diff
@@ -0,0 +1,25 @@
+use anyhow::Result;
+use async_trait::async_trait;
+
+use crate::{ConnectorStateV2, Properties, SourceMessage, SplitReader};
+
+/// [`DummySplitReader`] is a placeholder for source executor that is assigned no split. It will
+/// wait forever when calling `next`.
+#[derive(Clone, Debug)]
+pub struct DummySplitReader;
+
+#[async_trait]
+impl SplitReader for DummySplitReader {
+    async fn next(&mut self) -> Result<Option<Vec<SourceMessage>>> {
+        // the dead loop is intentional
+        #[allow(clippy::empty_loop)]
+        loop {}
```

Comment:
You may want something like this. https://docs.rs/futures/latest/futures/future/fn.pending.html